### DISCUSSION TITLE
Bug in updating the inline keyboard

### DISCUSSION THREAD

**Discussion by okaeiz** [kianahmadian/python-bale-bot#20] [Q&A] [answered]
Whenever an inline keyboard has to be edited and have a new caption and components (buttons), the caption gets edited but the buttons will not be updated until the user quits the conversation and re-enter it.

Here is the code:
```python
if callback.data == "help":
  new_caption = "This is the new caption"
  await callback.message.edit(text= new_caption,
                                      components=Components(
                  inline_keyboards=[InlineKeyboard("⤴️ back", callback_data="back")] ))
```
So in the main menu, if I click on "help", the caption will be edited to `new_caption` but the inline keyboard will not be updated until I quit the conversation with the bot and re-enter it.

**This problem only exists in **_Android_**** version of Bale Messenger; thus it is not reproducible in Windows and Linux web interfaces. Since the bot works fine in the web version, I think the problem is not with the code.

**Comment by kian-ahmadian** [comment, marked as answer]
### If this problem is only for Android versions, then it is definitely not a code problem.  Please make sure you have the latest version of Bale Messenger

**Reply by okaeiz** [reply]
This is what ChatGPT suggested:

**It seems like the issue you are experiencing is related to the caching mechanism of the Bale Messenger app on Android. Sometimes, the app caches the inline keyboard components and does not refresh them until the user leaves and rejoins the conversation.**

**Reply by okaeiz** [reply]
How can I remove the old keyboard components and send a new one? The only mentioning of removing components in the API reference could not help me.

**Remove Components**
`class bale.RemoveComponents`
This object shows a remove keyboard(s).

**Reply by kian-ahmadian** [reply]
**If you want to replace a new keyboard with the previous one, just send a new keyboard by sending a new message.**
